add Docker File and docker-compose

jeanvetorello opened the pull request:
refazendo o PR pois deu uma confusão minha aqui. 



Pessoal estou adicionando um docker file e um docker compose para ajudar o pessoal a utilizar varias contas, ao inves de utilizar maquinas virtuais, e tambem para facilitar o uso em vps.
para rodar bastar ter docker e docker compose instalado e ajustar o - VNC_PASSWORD=COLOCARUMASENHAAQUI. depois docker-compose up -d --build  essa image já tem o vnc instalado entao depois que ela rodar entre com o vnc no 127.0.0.1:5900 ,
tambem é possivel por http http://127.0.0.1:6080 e user root e senha que colocou no docker-compose.

apos isso abre o firefox instale metamask , add a rede da bsc e pronto. vai ter uma pasta com o repositorio atualizado no /root/Desktop
OBS: essa ultima atualizaçao do requirements.txt quebrou o build do container entao retirar o simpleaudio pois não tem audio mesmo .
TODO: acrescentar um volume onde guarde as info do firefox para que não se perca quando o container reinicar.


Update docker-compose.yml
ajuste para acessar o container por http

jeanvetorello committed:
add Docker File and docker-compose.

add Docker File and docker-compose
Pessoal estou adicionando um docker file e um docker compose para ajudar o pessoal a utilizar varias contas, ao inves de utilizar maquinas virtuais, e tambem para facilitar o uso em vps.
para rodar bastar ter docker e docker compose instalado e ajustar o - VNC_PASSWORD=COLOCARUMASENHAAQUI. depois docker-compose up -d --build essa image já tem o vnc instalado entao depois que ela rodar entre com o vnc no 127.0.0.1:5900 , coloca a senha que está la na variavel e pronto. vc está acessando remoto um linux,
localhost:6080  user root senha que mudou no HTTP_PASSWORD

atenção não tem criptografia http e o vnc tambem é sem criptografia. apenas testes locais não coloquem publico ao menos que vc saiba oque está fazendo.

apos isso abre o firefox instale metamask , add a rede da bsc e pronto. vai ter uma pasta com o repositorio atualizado no /root/Desktop
OBS: essa ultima atualizaçao do requirements.txt quebrou o build do container entao retirar o simpleaudio pois não tem audio mesmo .
TODO: acrescentar um volume onde guarde as info do firefox para que não se perca quando o contaer reinicar.


Update docker-compose.yml
ajuste para acessar o container por http

para acesso o container por http abrir o browser e acessar localhost:6080  user root senha que mudou no HTTP_PASSWORD


desculpe a confusão tinha feito uma bagunça no outro Pr agora esse tá so o necessário

jeanvetorello committed:
Merge pull request #1 from jeanvetorello/dockerimages

add Docker File and docker-compose.

fguide commented:
Bom trabalho, ficou bacana!

thiago-molive commented:
ja que n tem áudio acredito que a modificação do amigo que adicionou alerta pelo telegram possa ser interessante, enquanto o captcha não fica disponível

abanzy commented:
@jeanvetorello Eu tava pensando nisso, boa iniciativa, talvez seja interessante usar um alpine com Xfce como base para otimizar o resultado e escalabilidade. 
 Fica aí a referência https://github.com/edgelevel/alpine-xfce-vnc

jeanvetorello commented:
> @jeanvetorello Eu tava pensando nisso, boa iniciativa, talvez seja interessante usar um alpine com Xfce como base para otimizar o resultado e escalabilidade. Fica aí a referência https://github.com/edgelevel/alpine-xfce-vnc

sim mais leve , podemos tentar, mas fiz assim por conta  das depedencias que instalaram de boa , vou precisar procurar as dependencia no alpine.

uma das prioridade e criar um volume para que quando o container restart não se perca as configs do firefox.


hoje tenho uma vps linux só com o docker instaldo quando quero mais uma conta só subo mais um container , ficando bem mais simples. uma maquina na vps aguenta bastante contas pelo oque estava vendo.

jeanvetorello commented:
> ja que n tem áudio acredito que a modificação do amigo que adicionou alerta pelo telegram possa ser interessante, enquanto o captcha não fica disponível

sim fiz em repositorio local só seguir a dica que o pessoal deu la nos issue.

ricardofernandess commented:
Puts, que ideia boa parabéns!! Vai ficar mais leve do que rodar várias VM's pra multiaccount! 

Eu precisei adicionar o comando `sudo apt-get install -y python3-dev libasound2-dev` no Dockerfile pois dava erro em algo relacionado ao simpleaudio.

viniciusfontao commented:
só um ponto que também tive que fazer, meu PC tem 32gb ram e subi 2 containers, tava sendo utilizado 24GB de RAM e o processador tava em 90%
Achei uma config que resolveu isso:
Criar o arquivo .wslconfig em : %UserProfile%\.wslconfig
e adicionei essas config:
```
[wsl2]
memory=10GB
processors=4
```
Com essa config até o momento a % de uso de processador dos containers (total) está com 35% e o uso de RAM em 7GB

jeanvetorello committed:
Merge branch 'mpcabete:main' into main

jeanvetorello committed:
Merge branch 'mpcabete:main' into main

jeanvetorello committed:
Merge branch 'mpcabete:main' into main

far0fa commented:
vish nao tem um jeito mais facil? sou meio burro haeuhe

0bomb4 commented:
@far0fa  to na mesma, achei interessante mas nao faço a minima ideia por onde começar kk

ricardofernandess commented:
Pra quem não é dev fica complicado mesmo hehe.... A ideia é rodar em container, para isso precisa ter o docker e o docker-compose instalado na máquina. Se estiverem utilizando Windows instalem o Docker Desktop. A partir daí é só seguir as instruções do autor no primeiro post

jeanvetorello committed:
Merge branch 'mpcabete:main' into main

rzanca commented:
@jeanvetorello Estou testando o docker, ficou excelente, minha única dúvida seria, como subir mais de 1 conta, até o momento não conseguir subir 2 conteiners, usando o comando só aparece o bomb_bot_1

jeanvetorello commented:
> @jeanvetorello Estou testando o docker, ficou excelente, minha única dúvida seria, como subir mais de 1 conta, até o momento não conseguir subir 2 conteiners, usando o comando só aparece o bomb_bot_1

Ajuste seu docker compose coloando mais um service segue exemplo:
depois de o comando novamente docker-compose up -d

version: "3"
services:
  bot:
    build:
      context: .
    restart: unless-stopped
    environment:
      - VNC_PASSWORD=123mudar
      - HTTP_PASSWORD=123mudar
      - TZ=America/Sao_Paulo
      - RESOLUTION=1920x1080
    volumes:
       - /dev/shm:/dev/shm
    ports:
      - 5900:5900
      - 6080:80

  bot2:
    build:
      context: .
    restart: unless-stopped
    environment:
      - VNC_PASSWORD=123mudar
      - HTTP_PASSWORD=123mudar
      - TZ=America/Sao_Paulo
      - RESOLUTION=1920x1080
    volumes:
       - /dev/shm:/dev/shm
    ports:
      - 5901:5900
      - 6081:80

rzanca commented:
Valeu!!!

jeanvetorello committed:
Merge branch 'mpcabete:main' into main

jeanvetorello committed:
Merge branch 'mpcabete:main' into main

jeanvetorello committed:
Add files via upload

new connect wallet button

jeanvetorello committed:
merge grom main

jeanvetorello committed:
Revert "Merge branch 'mpcabete:main' into main"

This reverts commit 6f70c136c6dff966782b1e8e11b3b51a650924b9, reversing
changes made to 6ee6c508dcf0e7fbfd3aaec647712c49124e5d05.

jeanvetorello committed:
add connect wallet

jotapepinheiro commented:
```yml
version: "3"
services:
  bot-1:
    container_name: bot-1
    tty: true
    build:
      context: .
    restart: unless-stopped
    cap_add:
      #Google Chrome
      - SYS_ADMIN
    environment:
      - VNC_PASSWORD=123mudar
      - HTTP_PASSWORD=123mudar
      - TZ=America/Sao_Paulo
      - RESOLUTION=1920x1080
      - USER=ubuntu
    volumes:
       - /dev/shm:/dev/shm
    ports:
      - 5900:5900
      - 6080:80

  bot-2:
    container_name: bot-2
    tty: true
    build:
      context: .
    restart: unless-stopped
    cap_add:
      #Google Chrome
      - SYS_ADMIN
    environment:
      - VNC_PASSWORD=123mudar
      - HTTP_PASSWORD=123mudar
      - TZ=America/Sao_Paulo
      - RESOLUTION=1920x1080
      - USER=ubuntu
    volumes:
       - /dev/shm:/dev/shm
    ports:
      - 5901:5900
      - 6081:80
```

jotapepinheiro commented:
Arquivo makefile

```makefile
.PHONY: help
.DEFAULT_GOAL = help

## —— Docker 🐳  ———————————————————————————————————————————————————————————————
start-all: ## Iniciar todos os containers
	docker-compose up -d;

stop-all: ## Iniciar todos os containers
	docker-compose down;

rebuild-all: ## Rebuild em todos os containers
	docker-compose down && docker-compose up -d --build;

bot-1-rebuild: ## Rebuild bot-1
	docker-compose build --no-cache bot-1;

bot-2-rebuild: ## Rebuild bot-2
	docker-compose build --no-cache bot-2;

bot-1-shell: ## Acessar container bot-1
	docker container exec -it bot-1 bash;

bot-2-shell: ## Acessar container bot-2
	docker container exec -it bot-2 bash;

## —— SSH 🎶 ———————————————————————————————————————————————————————————————
bot-1-init: ## Enviar comando para bot 1
	docker-compose exec bot-1 sh -c ""

bot-2-init: ## Enviar comando para bot 2
	docker-compose exec bot-2 sh -c ""

## —— Outros 🛠️️ ———————————————————————————————————————————————————————————————

help: ## Lista de commandos
	@grep -E '(^[a-zA-Z_-]+:.*?##.*$$)|(^##)' $(MAKEFILE_LIST) \
	| awk 'BEGIN {FS = ":.*?## "}; {printf "\033[32m%-24s\033[0m %s\n", $$1, $$2}' \
	| sed -e 's/\[32m## /[33m/' && printf "\n"

```

BrunoLagoa commented:
> > @jeanvetorello Eu tava pensando nisso, boa iniciativa, talvez seja interessante usar um alpine com Xfce como base para otimizar o resultado e escalabilidade. Fica aí a referência https://github.com/edgelevel/alpine-xfce-vnc
> 
> sim mais leve , podemos tentar, mas fiz assim por conta das depedencias que instalaram de boa , vou precisar procurar as dependencia no alpine.
> 
> uma das prioridade e criar um volume para que quando o container restart não se perca as configs do firefox.
> 
> hoje tenho uma vps linux só com o docker instaldo quando quero mais uma conta só subo mais um container , ficando bem mais simples. uma maquina na vps aguenta bastante contas pelo oque estava vendo.

conseguiu fazer o container não perder as configs do firefox?

jeanvetorello committed:
Merge branch 'mpcabete:main' into main

jeanvetorello committed:
Revert "optmized sign button image"

This reverts commit 81df9839313999d0a57dce6fc593a3570155edf8.

jeanvetorello committed:
Revert "new slider logic"

This reverts commit 40beeffe6d37141ce5aa327726d4be79f6b68452.

Denver3D commented:
any video tutorial ?

jhonsstn commented:
![image](https://user-images.githubusercontent.com/33515228/146658741-72f244aa-2ceb-4ad5-98fa-f8fce57a8ce5.png)

2 contas rodando em 2 containers em uma VM no Google de 6vCPU e 8GB RAM. Usando o OS otimizado para containers deles mesmo. O Webgl da uma puxada de CPU boa. Mas ta indo razoavelmente bem. Economizar pro trial durar mais. Como esta o setup de vcs?

rzanca commented:
> ![image](https://user-images.githubusercontent.com/33515228/146658741-72f244aa-2ceb-4ad5-98fa-f8fce57a8ce5.png)
> 
> 2 contas rodando em 2 containers em uma VM no Google de 6vCPU e 8GB RAM. Usando o OS otimizado para containers deles mesmo. O Webgl da uma puxada de CPU boa. Mas ta indo razoavelmente bem. Economizar pro trial durar mais. Como esta o setup de vcs?

Qual a configuração que utilizou para criar a VM? Estou rodando em um Notebook i5-8265u com 16gb ram, mas tá explodindo a máquina com 2 containers... rsrs

jeanvetorello commented:
tenho uma vps 4 cpu e 8gb de RAM rodo 3 container, mas fica sempre no 100% . pelo meus calculo cada conta é 1 vcpu + uns 3 GB RAM

jeanvetorello committed:
Revert "slider update"

This reverts commit ad98d2f82ac274eff2d54458ddec92815a27bfad.

jeanvetorello committed:
Merge branch 'main' of https://github.com/mpcabete/bombcrypto-bot into mpcabete-main

jeanvetorello committed:
Merge branch 'mpcabete-main' into main

rzanca commented:
> tenho uma vps 4 cpu e 8gb de RAM rodo 3 container, mas fica sempre no 100% . pelo meus calculo cada conta é 1 vcpu + uns 3 GB RAM

Usou qual OS na criação?
Eu tava pensando em pegar o 8vcpu/16gb pelas minhas contas daria 4 meses free. Com o atual crédito que liberaram.

fguide commented:
Eu acho docker uma solução boa, mas pra esse bot não funciona eficientemente, acho que só uma VM com várias instancias do VNC ja resolveria o problema sem gargalos.

Pelo menos dessa forma você nao atribui as vCPUs a containers fixoes e faz o processamento ser mais eficiente.

Não adianta esperar milagre tb, HTML5 + Unity, nenhum navegador, sistema ou solucao é otimizado suficiente pra rodar com poucos recursos, entao o melhor é saber distribuir da melhor forma.

jeanvetorello commented:
> Eu acho docker uma solução boa, mas pra esse bot não funciona eficientemente, acho que só uma VM com várias instancias do VNC ja resolveria o problema sem gargalos.
> 
> Pelo menos dessa forma você nao atribui as vCPUs a containers fixoes e faz o processamento ser mais eficiente.
> 
> Não adianta esperar milagre tb, HTML5 + Unity, nenhum navegador, sistema ou solucao é otimizado suficiente pra rodar com poucos recursos, entao o melhor é saber distribuir da melhor forma.

tambem acho bem estranho esse joguindo consumir tanto CPU e memoria se eu limito no docker menos cpu e memoria o firefox crash  toda hora. mas enfim não tem milagre.

jeanvetorello commented:
> > tenho uma vps 4 cpu e 8gb de RAM rodo 3 container, mas fica sempre no 100% . pelo meus calculo cada conta é 1 vcpu + uns 3 GB RAM
> 
> Usou qual OS na criação? Eu tava pensando em pegar o 8vcpu/16gb pelas minhas contas daria 4 meses free. Com o atual crédito que liberaram.

ubunntu 20.04 só com o docker e docker compose instalado.


com essa sua config daria para cololocar uns 6 contas nela.  eu colocaria e monitorava para ver.

fguide commented:
> > Eu acho docker uma solução boa, mas pra esse bot não funciona eficientemente, acho que só uma VM com várias instancias do VNC ja resolveria o problema sem gargalos.
> > Pelo menos dessa forma você nao atribui as vCPUs a containers fixoes e faz o processamento ser mais eficiente.
> > Não adianta esperar milagre tb, HTML5 + Unity, nenhum navegador, sistema ou solucao é otimizado suficiente pra rodar com poucos recursos, entao o melhor é saber distribuir da melhor forma.
> 
> tambem acho bem estranho esse joguindo consumir tanto CPU e memoria se eu limito no docker menos cpu e memoria o firefox crash toda hora. mas enfim não tem milagre.

Sim, esse é um "problema" sem solução, nós temos muitas opções de gerenciadores de janela, navegadores, sistemas, etc que são otimizados e super leves, o raspbian é um exemplo, porém a "internet" de hoje é pesada e inviável com recursos limitados.

Eu sugiro fazer um teste apenas com o VNC em várias instâncias, cada uma tem o seu display proprio e o bot roda individualmente.

Outra ideia se for máquina *nix física (ou se tiver acesso ao console do virtualizador), pode usar vários tty diferentes, mas dai exige carregar um ambiente gráfico para cada tty, se for alo mais leve como openbox ou i3wm seria melhor.

rzanca commented:
> > > Eu acho docker uma solução boa, mas pra esse bot não funciona eficientemente, acho que só uma VM com várias instancias do VNC ja resolveria o problema sem gargalos.
> > > Pelo menos dessa forma você nao atribui as vCPUs a containers fixoes e faz o processamento ser mais eficiente.
> > > Não adianta esperar milagre tb, HTML5 + Unity, nenhum navegador, sistema ou solucao é otimizado suficiente pra rodar com poucos recursos, entao o melhor é saber distribuir da melhor forma.
> > 
> > 
> > tambem acho bem estranho esse joguindo consumir tanto CPU e memoria se eu limito no docker menos cpu e memoria o firefox crash toda hora. mas enfim não tem milagre.
> 
> Sim, esse é um "problema" sem solução, nós temos muitas opções de gerenciadores de janela, navegadores, sistemas, etc que são otimizados e super leves, o raspbian é um exemplo, porém a "internet" de hoje é pesada e inviável com recursos limitados.
> 
> Eu sugiro fazer um teste apenas com o VNC em várias instâncias, cada uma tem o seu display proprio e o bot roda individualmente.
> 
> Outra ideia se for máquina *nix física (ou se tiver acesso ao console do virtualizador), pode usar vários tty diferentes, mas dai exige carregar um ambiente gráfico para cada tty, se for alo mais leve como openbox ou i3wm seria melhor.

Se o jogo rodasse tranquilamente em 1vcpu + 2gb, ficaria muito mais viável abrir separado mesmo. O custo total seria reduzido. Porém não sei como ficaria.

elequiel commented:
Pessoal, sobre o TODO do volume eu fiz da seguinte maneira:
E também limitei o consumo de memoria e cpu de cada service

```yaml
version: "3"
services:
  bot:
    build:
      context: .
    restart: unless-stopped
    environment:
      - VNC_PASSWORD=4516
      - HTTP_PASSWORD=4516
      - TZ=America/Sao_Paulo
      - RESOLUTION=1920x1080
    volumes:
       - /dev/shm:/dev/shm
       - ./PROFILES/MOZILLA/bot1:/root/.mozilla/ 
    mem_limit: 1024m
    cpus: 1
    ports:
      - 5900:5900
      - 6080:80
```

Com isso não será necessário configurar tudo sempre reiniciar o docker compose

jeanvetorello commented:
> Pessoal, sobre o TODO do volume eu fiz da seguinte maneira: E também limitei o consumo de memoria e cpu de cada service
> 
> ```yaml
> version: "3"
> services:
>   bot:
>     build:
>       context: .
>     restart: unless-stopped
>     environment:
>       - VNC_PASSWORD=4516
>       - HTTP_PASSWORD=4516
>       - TZ=America/Sao_Paulo
>       - RESOLUTION=1920x1080
>     volumes:
>        - /dev/shm:/dev/shm
>        - ./PROFILES/MOZILLA/bot1:/root/.mozilla/ 
>     mem_limit: 1024m
>     cpus: 1
>     ports:
>       - 5900:5900
>       - 6080:80
> ```
> 
> Com isso não será necessário configurar tudo sempre reiniciar o docker compose

Top mano vlw pela colaboração.

rzanca commented:
> Pessoal, sobre o TODO do volume eu fiz da seguinte maneira:
> E também limitei o consumo de memoria e cpu de cada service
> 
> ```yaml
> version: "3"
> services:
>   bot:
>     build:
>       context: .
>     restart: unless-stopped
>     environment:
>       - VNC_PASSWORD=4516
>       - HTTP_PASSWORD=4516
>       - TZ=America/Sao_Paulo
>       - RESOLUTION=1920x1080
>     volumes:
>        - /dev/shm:/dev/shm
>        - ./PROFILES/MOZILLA/bot1:/root/.mozilla/ 
>     mem_limit: 1024m
>     cpus: 1
>     ports:
>       - 5900:5900
>       - 6080:80
> ```
> 
> Com isso não será necessário configurar tudo sempre reiniciar o docker compose
> 

Como ficou o sistema rodando dentro dessa limitação? Aqui estava insuportável.
Utilizando 3CPU/10GB para 3 Containers.

Na VPS tive problemas por conta do suporte a WEBGL. E depois estive com pouco tempo para verificar.

Mas se rodar com menos recurso. Ajudaria muito.

jeanvetorello commented:
> 

aqui eu rodo 3 container com 4vcpu e 8GB de ram , mas sem limitaçao.

quando coloco limitaçao o firefox da muito crash. 

vou tentar dar uma olhada nisso.

fguide commented:
Desligar a aceleração de hardware no firefox ajuda com VMs sem GPU. Não desative o webgl, só aceleração 3D.

elequiel commented:
> 

to usando 2 containers por enquanto mas não ta crashando muito não, desativar a aceleracao como o amigo comentou pra v se melhora o consumo de recursos

jeanvetorello committed:
Update docker-compose.yml

ajustes no docker-compose para salvar as configs do firefox denntro do volume , ajustado o docker compose para 2 contianer. 

obs : acompanhar se o firefox vai dar crash pela quantidade de memoria CPu reservado.

jhonsstn commented:
> Eu acho docker uma solução boa, mas pra esse bot não funciona eficientemente, acho que só uma VM com várias instancias do VNC ja resolveria o problema sem gargalos.
> 
> Pelo menos dessa forma você nao atribui as vCPUs a containers fixoes e faz o processamento ser mais eficiente.
> 
> Não adianta esperar milagre tb, HTML5 + Unity, nenhum navegador, sistema ou solucao é otimizado suficiente pra rodar com poucos recursos, entao o melhor é saber distribuir da melhor forma.

Saberia informar algum artigo de referencia para criar varias instancias de VNC?

fguide commented:
> > Eu acho docker uma solução boa, mas pra esse bot não funciona eficientemente, acho que só uma VM com várias instancias do VNC ja resolveria o problema sem gargalos.
> > Pelo menos dessa forma você nao atribui as vCPUs a containers fixoes e faz o processamento ser mais eficiente.
> > Não adianta esperar milagre tb, HTML5 + Unity, nenhum navegador, sistema ou solucao é otimizado suficiente pra rodar com poucos recursos, entao o melhor é saber distribuir da melhor forma.
> 
> Saberia informar algum artigo de referencia para criar varias instancias de VNC?

Não tenho nenhum fácil aqui, mas o processo é simples:

Configure o VNC
Quando vc executa o vncserver vc precisa informar o display, por exemplo: 'vncserver :1'  abrirá uma sessão no display 1, na configuracao padrao na porta 5901.
Se vc executar 'vncserver :2' vai abrir no display 2 e porta 5901 e assim sucessivamente.

Dai você pode ou acessar cada sessão normalmente pleo VNC usando as portas 590X ou mesmo abrir uma sessao SSH e exportar o display para executar tudo de um terminal só, ex:

export DISPLAY=:1
python index.py &
export DISPLAY=:2
python index.py &

E por ai vai, com VNC tem várias opções, inclusive usar o noVNC para acessar via html.

jeanvetorello commented:
alguem sabe onde ficam os arquivos do chrome ou do brave to pensando em colocalos no container aqui para ver o desempenho se fica melhor.

jhonsstn commented:
> Não tenho nenhum fácil aqui, mas o processo é simples:
> 
> Configure o VNC Quando vc executa o vncserver vc precisa informar o display, por exemplo: 'vncserver :1' abrirá uma sessão no display 1, na configuracao padrao na porta 5901. Se vc executar 'vncserver :2' vai abrir no display 2 e porta 5902 e assim sucessivamente.
> 
> Dai você pode ou acessar cada sessão normalmente pleo VNC usando as portas 590X ou mesmo abrir uma sessao SSH e exportar o display para executar tudo de um terminal só, ex:
> 
> export DISPLAY=:1 python index.py & export DISPLAY=:2 python index.py &
> 
> E por ai vai, com VNC tem várias opções, inclusive usar o noVNC para acessar via html.

Opa. É mais do que o suficiente. Encontrei muito conteúdo relacionado a x display. Obrigado pelo norte.

jeanvetorello committed:
Merge branch 'mpcabete:main' into main

jhonsstn commented:
Será que o desempenho da maquina, e por consequência o framerate do game, pode alterar o farm? Por pelo menos visivelmente os bonecos parecerem se mover mais lentamente e com travamentos.

arthuravila26 commented:
Uma dica sobre o Dockerfile. Não é uma boa pratica manter as linhas do Dockerfile muito longas, isso pode dificultar um pouco a leitura de outros Devs. Eu faria algo parecido com isso para ficar mais legivel:
`FROM dorowu/ubuntu-desktop-lxde-vnc

RUN apt-get update && sudo apt-get install python3-pyqt5 -y \
    && sudo apt-get install git -y \
    && sudo apt-get install pyqt5-dev-tools -y \
    && sudo apt-get install qttools5-dev-tools -y \
    && apt-get install xclip -y \
    && apt-get install python3-pip -y \
    && apt-get install python3-tk python3-dev -y \
    && pip install pyautogui ; pip install python-xlib

#install DEPs
RUN git clone https://github.com/mpcabete/bombcrypto-bot.git /root/Desktop/bombcrypto-bot \
    && pip install -r /root/Desktop/bombcrypto-bot/requirements.txt`

Desta forma, fica mais fácil para identificar oq esta sendo instalado e etc.
Btw, ótimo trabalho, ficou bem legal mesmo =D

jeanvetorello committed:
from main

jeanvetorello committed:
ajust files

jeanvetorello committed:
Update Dockerfile

remove firefox and google chrome.
install brave browser for more privacy.

jeanvetorello committed:
Update docker-compose.yml

ajust volumes and resources limits

wenked commented:
O brave não abre aqui na vm

jeanvetorello commented:
> O brave não abre aqui na vm

aqui abre normal, executou o compose exatamente igual ?

wenked commented:
> > O brave não abre aqui na vm
> 
> aqui abre normal, executou o compose exatamente igual ?

Sim , executei exatamente igual

abanzy commented:
> > O brave não abre aqui na vm
> 
> aqui abre normal, executou o compose exatamente igual ?

Brave nao abriu aqui tambem, voltei para o firefox.

 E reparei que o uso de memora chegou a dobrar ao longo de 20 horas (so nao sei se foi devido à swap que adicionei e ta acumulando lixo)

wenked commented:
Eu consegui abrir o brave usando esse comando ->` brave-browser google.com --disable-dev-shm-usage`

giovanniglsbl commented:
> Eu consegui abrir o brave usando esse comando ->` brave-browser google.com --disable-dev-shm-usage`

Salvou !!!!
